sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -26,43 +26,64 @@ import org.apache.spark.sql.expressions.Aggregator
 // This file defines internal implementations for aggregators.
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+class TypedSumDouble[IN](val f: IN => Double)
+  extends Aggregator[IN, java.lang.Double, java.lang.Double] {
+
+  override def zero: java.lang.Double = null
+  override def reduce(b: java.lang.Double, a: IN): java.lang.Double =
```

Comment:
I'm afraid this may have performance regression because of boxing. Actually I think it's fine to have the typed operations have different semantic from SQL. @rxin what do you think? Can typed sum return `0` for empty input?

Reply:
OK let's follow the SQL semantic and deal with performance problem later. @setjet sorry for the late.